app/workers/terms_of_service_update_worker.rb

```diff
@@ -6,9 +6,15 @@ class TermsOfServiceUpdateWorker
   def perform
     last_updated = Time.at(User.terms_last_updated_at)
     updated_time = Time.now
-    User.where(is_active: true).where('email IS NOT NULL').where('last_received_terms_email_at < ?', last_updated).find_each do |u|
-      UpdatedTermsMailer.delay({ retry: 0, queue: 'terms_mailer' }).notify(u.email, u.name)
-      u.update_columns(last_received_terms_email_at: updated_time)
+    # Based on our AWS SES account (Maximum send rate: 200 emails per second) I set a batch size 200 and do a sleep 1
+    User.where(is_active: true).where('email IS NOT NULL')
+    .where('last_received_terms_email_at < ?', last_updated)
+    .find_in_batches(:batch_size => 200) do |users|
+      users.each do |u|
+        UpdatedTermsMailer.delay({ retry: 1, queue: 'terms_mailer' }).notify(u.email, u.name)
+        u.update_columns(last_received_terms_email_at: updated_time)
+      end
+      sleep 1
```

Comment:
Sawy, I think it's safer to use 2 seconds just to account for the time spent until and after the job runs.

Reply:
Done